certbot-ci/certbot_integration_tests/utils/acme_server.py

```diff
@@ -63,12 +63,14 @@ def __init__(self, acme_server: str, nodes: List[str], http_proxy: bool = True,
         self._processes: List[subprocess.Popen] = []
         self._stdout = sys.stdout if stdout else open(os.devnull, 'w') # pylint: disable=consider-using-with
         self._dns_server = dns_server
-        self._http_01_port = http_01_port
         self._preterminate_cmds_args: List[Tuple[Tuple[Any, ...], Dict[str, Any]]] = []
-        if http_01_port != DEFAULT_HTTP_01_PORT:
-            if self._acme_type != 'pebble' or self._proxy:
-                raise ValueError('setting http_01_port is not currently supported '
-                                  'with boulder or the HTTP proxy')
+        self._http_01_port = BOULDER_HTTP_01_PORT if self._acme_type == 'boulder' \
+                             else DEFAULT_HTTP_01_PORT
+        if http_01_port:
+            if (self._acme_type == 'pebble' and self._proxy) or self._acme_type == 'boulder':
```

Comment:
Why change this away from `self._acme_type != 'pebble' or self._proxy` (or `self._acme_type == 'boulder' or self._proxy`)?

Reply:
Oops 😆. I found it difficult to visualize the "truth table" of the original, but it looks like I ended up with the same thing effectively. I am going to merge to avoid another review cycle, but I take your point.